src/Common/UpgradeNotification/UpgradeNotificationHelper.cs

```diff
@@ -0,0 +1,259 @@
+            }
+        }
+
+        private void RefreshVersionInfo(string loadModuleNames)
+        {
+            this.versionDict = LoadHigherAzVersions(loadModuleNames);
+            if (!VersionsAreFreshed())
+            {
+                return;
+            }
+            string content = JsonConvert.SerializeObject(this.versionDict);
+            using (StreamWriter sw = new StreamWriter(new FileStream(AzVersionCacheFile, FileMode.Create, FileAccess.Write, FileShare.None)))
+            {
+                sw.Write(content);
+            }
+        }
+
+        private bool VersionsAreFreshed()
+        {
+            return versionDict != null && versionDict.Count > 0;
+        }
+
+        private string GetModuleLatestVersion(string moduleName)
+        {
+            string defaultVersion = "0.0.0";
+            if (!VersionsAreFreshed())
+            {
+                return defaultVersion;
+            }
+            return versionDict.ContainsKey(moduleName) ? versionDict[moduleName] : defaultVersion;
+        }
+
+        private bool HasHigherVersion(string moduleName, string currentVersion)
+        {
+            if (!VersionsAreFreshed())
+            {
+                return false;
+            }
+            try
+            {
+                Version currentVersionValue = Version.Parse(currentVersion);
+                Version latestVersionValue = Version.Parse(versionDict[moduleName]);
+                return latestVersionValue > currentVersionValue;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        private bool HasHigherMajorVersion(string moduleName, string currentVersion)
+        {
+            if (!VersionsAreFreshed())
+            {
+                return false;
+            }
+            try
+            {
+                Version currentVersionValue = Version.Parse(currentVersion);
+                Version latestVersionValue = Version.Parse(versionDict[moduleName]);
+                return latestVersionValue.Major > currentVersionValue.Major;
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+        }
+
+        private static Dictionary<string, string> LoadHigherAzVersions(string moduleName)
+        {
+            Dictionary<string, string> versionDict = new Dictionary<string, string>();
+
+            string findModuleCmdlet = GetCmdletForFindModule();
+            findModuleCmdlet += " -Name " + moduleName + " | Select-Object Name, Version";
+
+            var outputs = ExecutePSScript<PSObject>(findModuleCmdlet);
+            foreach (PSObject obj in outputs)
+            {
+                versionDict[obj.Properties["Name"].Value.ToString()] = obj.Properties["Version"].Value.ToString();
+            }
+            return versionDict;
+        }
+
+        private static string GetCmdletForUpdateModule()
```

Comment:
Perhaps we could recommend that customers utilize `Az.Tools.Installer` to manage their Az modules.

Reply:
It's an option. But Az.Tools.Installer is not included in Az for now. That's why we use "PowerShellGet" cmdlets instead of Az.Tools.Installer.